File: simpleadapter/src/main/java/com/github/deckyfx/simpleadapter/ExpandableAdapterDataSet.java
package com.github.deckyfx.simpleadapter;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import java.util.RandomAccess;

/**
 * Created by decky on 5/21/15.
 */
public class ExpandableAdapterDataSet<E extends AdapterGroupItem, T extends BaseItem> extends ArrayList<E> implements List<E>, RandomAccess, Cloneable, Serializable {
    public ExpandableAdapterDataSet<E, T> find(CharSequence constraint) {
        ExpandableAdapterDataSet<E, T> results = new ExpandableAdapterDataSet<E, T>();
        for (int i = 0; i < this.size(); i++) {
            if (this.get(i).testFilter(constraint)) {
                results.add(this.get(i));
            }
        }
        return results;
    }

    @Override
    public E get(int i) {
        return super.get(i);
    }


    public <T> E get(int i, boolean f) {
        return super.get(i);
    }

    @Override
    public int size() {
        return super.size();
    }
}

File: simpleadapter-kotlin/src/main/java/com/github/simpleadapter/ExpandableAdapterDataSet.kt
package com.itp.android.simpleadapter_kotlin

import java.io.Serializable
import java.util.ArrayList
import java.util.RandomAccess

/**
 * Created by decky on 5/21/15.
 */
class ExpandableAdapterDataSet<E : AdapterGroupItem<*>, T : AdapterItem> : ArrayList<E>(), List<E>, RandomAccess, Cloneable, Serializable {
    fun find(constraint: CharSequence): ExpandableAdapterDataSet<E, T> {
        val results = ExpandableAdapterDataSet<E, T>()
        for (i in 0..this.size - 1) {
            if (this[i].testFilter(constraint)) {
                results.add(this[i])
            }
        }
        return results
    }

    override fun get(i: Int): E {
        return super.get(i)
    }


    operator fun <T> get(i: Int, f: Boolean): E {
        return super.get(i)
    }

    override fun clone(): Any {
        return super<ArrayList>.clone();
    }
}
